lib/Runtime/Base/FunctionBody.h

```diff
@@ -1159,10 +1195,49 @@ namespace Js
     //
     class FunctionProxy : public FunctionInfo
     {
+    public:
+        typedef RecyclerWeakReference<DynamicType> FunctionTypeWeakRef;
+        typedef JsUtil::List<FunctionTypeWeakRef*, Recycler, false, WeakRefFreeListedRemovePolicy> FunctionTypeWeakRefList;
+
     protected:
         FunctionProxy(JavascriptMethod entryPoint, Attributes attributes, int nestedCount, int derivedSize,
             LocalFunctionId functionId, ScriptContext* scriptContext, Utf8SourceInfo* utf8SourceInfo, uint functionNumber);
         DEFINE_VTABLE_CTOR_NO_REGISTER(FunctionProxy, FunctionInfo);
+
+        enum AUXPTRS : uint8 {
```

Comment:
Consider using a C++11 enum class instead so that the names are not polluting the FunctionProxy class scope and do not need to have e_ prepended to them to differentiate them (since the enum class will require scoping the enumerators with the enum class name).

This would allow the names to be inline with our coding conventions.